tcms/testruns/static/testruns/js/get.js

```diff
@@ -120,15 +120,24 @@ function renderAdditionalInformation(testExecutions, testExecutionCaseIds) {
             const listGroupItem = $(`.test-execution-${testExecution.id}`)
             listGroupItem.find('.test-execution-priority').html(testCase.priority)
             listGroupItem.find('.test-execution-category').html(testCase.category)
+            listGroupItem.find('.test-execution-text').html(testCase.text)
+            listGroupItem.find('.test-execution-notes').append(testCase.notes)
 
             const isAutomatedElement = listGroupItem.find('.test-execution-automated')
             const isAutomatedIcon = testCase.is_automated ? 'fa-cog' : 'fa-thumbs-up'
             const isAutomatedAttr = testCase.is_automated ? isAutomatedElement.data('automated') : isAutomatedElement.data('manual')
             isAutomatedElement.addClass(isAutomatedIcon)
             isAutomatedElement.attr('title', isAutomatedAttr)
 
-            jsonRPC('TestExecution.get_links', { 'execution_id': testExecution.id, 'is_defect': true }, bugs => {
-                listGroupItem.find('.test-execution-bugs-count').html(bugs.length)
+            jsonRPC('TestExecution.get_links', { 'execution_id': testExecution.id }, links => {
+                const bugCount = links.filter(link => link.is_defect).length;
+                listGroupItem.find('.test-execution-bugs-count').html(bugCount)
+
+                const ul = listGroupItem.find('.test-execution-hyperlinks')
+                links.forEach(link => {
+                    const icon = link.is_defect ? `<span class="fa fa-bug"></span>` : ''
```

Comment:
Bug icon is not showing tooltip, see for example TestCase page. You can actually try reusing the widget from that page here (will need some modification) b/c the existing widget is very similar to other such widgets which support adding and deleting items.

You will have to make sure the TC page is still read-only but for TE details you can reuse the entire add/delete/presentational functionality which will save you some duplication when you start implementing the missing add url/report bug actions.

Reply:
You can start with a read-only shared widget across both pages and then figure out how to make it R/W where necessary.